ml-algorithms/src/main/java/org/opensearch/ml/engine/tools/CatIndexTool.java

```diff
@@ -73,126 +86,210 @@ public Object parse(Object o) {
 
     @Override
     public <T> void run(Map<String, String> parameters, ActionListener<T> listener) {
-        List<String> indexList = gson.fromJson(parameters.get("indices"), List.class);
-        String[] indices = parameters.containsKey("indices")? indexList.toArray(new String[0]) : new String[]{};
+        // TODO: This logic exactly matches the OpenSearch _cat/indices REST action. If code at
```

Comment:
>This logic exactly matches the OpenSearch _cat/indices REST action.

If that's true, what's the issue if we invoke this rest api, get the result and perform necessary post processing and send that to the llm? 

Sorry I might not have enough context on this.

Reply:
If we want to add more functionalities such as _cat/shards, _cat/nodes, _cat/master, _cat/allocation, _cat/segments, _cat/transform, etc., will we continue to maintain separate code bases?

Reply:
Hi Dhrubo, currently we don't have a way to directly call rest api in plugin, as the plugin doesn't know how to provide credentials needed in rest api. So Dan provided some solutions here we can address them later. I don't think it is a blocking issue for these kind of tools development for now.

Reply:
Thanks @jngz-es for the clarification. I assume we can't invoke those APIs through transport layer either?

Reply:
@dhrubo-os we are invoking thru transport, the problem is that there is a lot of logic in between REST call and the transport.  We either have to duplicate the logic or find a common place to put it.

Reply:
@dhrubo-os to be more clear there is no transport API exactly corresponding to `_cat/indices`.  That REST API calls 4 transport APIs: settings to use as an index filter, and health (for that one stat), metadata (for uuid), and stats (for everything else), and then combine all the return values.  

In an ideal world we use the existing code, either by exposing currently private methods or by moving them to a public util class or by creating a new "parent" transport action above the logic that calls the other actions, etc.

Reply:
Thanks for the clarification @dbwiddis . Approving the PR.